src/pypromice/process/join_l3.py

```diff
@@ -0,0 +1,282 @@
+        df = df.set_index('timestamp')
+        ds = df.to_xarray()
+        ds.attrs = meta   
+    return ds
+
+
+def loadArr(infile):
+    if infile.split('.')[-1].lower() in 'csv':
+        with open(infile) as f:
+            text_splitted = f.read().splitlines()
+            first_char =  text_splitted[0][0]
+            
+        if first_char != '#':
+            df = pd.read_csv(infile)
+            df['time'] = pd.to_datetime(df['time']).dt.tz_localize(None)
+            df = df.set_index('time')
+            ds = xr.Dataset.from_dataframe(df)
+        else:
+            ds = readNead(infile)
+        f.close()
+    elif infile.split('.')[-1].lower() in 'nc':
+        ds = xr.open_dataset(infile)
+        for varname in ds.variables:
+            if 'encoding' in ds[varname].attrs:
+                del ds[varname].attrs['encoding']
+    
+    try:
+        name = ds.attrs['station_name'] 
+    except:
+        name = infile.split('/')[-1].split('.')[0].split('_hour')[0].split('_10min')[0]
+        
+    print(f'{name} array loaded from {infile}')
+    return ds, name
+
+
+def gcnet_postprocessing(l3):
+    file_path = pkg_resources.resource_stream('pypromice', 'ressources/variable_aliases_GC-Net.csv')
+     
+    var_name = pd.read_csv(file_path)
+    var_name = var_name.set_index('old_name').GEUS_name
+    msk = [v for v in var_name.index if v in l3.data_vars]
+    var_name = var_name.loc[msk].to_dict()
+    
+    l3['TA1'] =  l3['TA1'].combine_first(l3['TA3'])
+    l3['TA2'] =  l3['TA2'].combine_first(l3['TA4'])
+    
+    l3=l3.rename(var_name)
+    l3=l3.rename({'timestamp':'time'})
+    return l3
+
+# will be used in the future
+# def aligning_surface_heights(l3m, l3):
+    # df_aux['z_surf_combined'] = \
+    #     df_aux['z_surf_combined'] \
+    #         - df_aux.loc[df_aux.z_surf_combined.last_valid_index(), 'z_surf_combined'] \
+    #          + df_v6.loc[df_v6.z_surf_combined.first_valid_index(), 'z_surf_combined'] 
+    
+    # if s == 'Swiss Camp 10m':        
+    #     df.loc[:df.HS_combined.first_valid_index(), 'HS_combined'] = \
+    #         df2.loc[:df.HS_combined.first_valid_index(), 'HS_combined'] \
+    #             - df2.loc[df2.HS_combined.last_valid_index(), 'HS_combined'] \
+    #              + df.loc[df.HS_combined.first_valid_index(), 'HS_combined'] 
+
+                    
+    # df.loc[df.HS_combined.diff()==0,'HS_combined'] = np.nan
+    
+    # fit = np.polyfit(df.loc[df.HS_combined.notnull(),:].index.astype('int64'),  
+    #                    df.loc[df.HS_combined.notnull(),'HS_combined'], 1)
+    # fit_fn = np.poly1d(fit)
+    
+    # df['HS_combined'] = df['HS_combined'].values \
+    #     - fit_fn(
+    #         df_in.loc[[df_in.z_surf_combined.first_valid_index()],:].index.astype('int64')[0]
+    #     )  +  df_in.loc[df_in.z_surf_combined.first_valid_index(), 'z_surf_combined']
+    # return l3m
+    
+def build_station_dict(config_folder):
```

Comment:
Why not creating a dictionary where the station configurations are grouped with respect to sites.

```python
def build_site_configurations(config_folder) -> Dict[str, List[Dict]]:
    """
    Build a dictionary with a list of station configurations for each site.

    Parameters
    ----------
    config_folder: str
        Path to the folder containing the station configuration toml files

    Returns
    -------

    """
    site_stations = dict()

    for station_config_path in Path(config_folder).glob('*.toml'):
        with station_config_path.open() as fp:
            station_config = toml.load(fp)
            site_id = station_config.get(site_id)
            if site_id:
                site_stations[site_id] = site_stations.get(site_id, [])
                site_stations[site_id].append(station_config)
    return site_stations
```

Note: I noticed that the station config toml files in aws_l0 started with a top level key equal to the station id. I think you should avoid this key since one toml file corresponds to one station.

```toml
stid = "EGP"
station_site = "EGP"
project = "GC-Net"
# ...
```
Without the `[EGP]` at the beginning.

Reply:
> I noticed that the station config toml files in aws_l0 started with a top level key equal to the station id. I think you should avoid this key since one toml file corresponds to one station.

I prepared an update following your suggestion: https://geusgitlab.geus.dk/glaciology-and-climate/promice/aws-l0/-/commit/30d15e6712656183c86cc2f0bc9daa52c579678d

Reply:
I am now building a `station_list` from an input `site`:

```

def build_station_list(config_folder: str, target_station_site: str):
    """
    Get a list of unique station IDs (stid) for a given station site.

    Parameters
    ----------
    config_folder : str
        Path to the folder containing the station configuration TOML files.
    target_station_site : str
        The station site to filter the station IDs by.

    Returns
    -------
    list
        A list of unique station IDs that have the specified station site.
    """
    unique_stids = []  # Initialize an empty list to store unique station IDs
    
    for filename in os.listdir(config_folder):
        if filename.endswith(".toml"):
            file_path = os.path.join(config_folder, filename)
            
            with open(file_path, 'r') as file:
                data = toml.load(file)  # Load the TOML file
                station_site = data.get("station_site")  # Get the station site
                stid = data.get("stid")  # Get the station ID
                
                # Check if the station site matches the target and stid is unique
                if station_site == target_station_site and stid and stid not in unique_stids:
                    unique_stids.append(stid)  # Add the stid to the list if unique
    
    return unique_stids

    
def join_l3():
    args = parse_arguments_joinl3()
    logging.basicConfig(
        format="%(asctime)s; %(levelname)s; %(name)s; %(message)s",
        level=logging.INFO,
        stream=sys.stdout,
    )
    
    station_list = build_station_list(args.config_folder, args.site)

    l3m = xr.Dataset()
    for stid in  station_list:
        ...
```

Reply:
As mentioned in https://github.com/GEUS-Glaciology-and-Climate/pypromice/pull/252#discussion_r1645752832, I suggest you return a list of station configuration dicts (`data`) instead of only the the uuids.

Reply:
See comment above. I do as your suggest and load all the info from the toml file.